src/legacy/core_plugins/kibana/public/management/sections/settings/lib/get_val_type.ts

```diff
@@ -22,7 +22,11 @@
  * @param {?} current value of the setting
  * @returns {string} the type to use for determining the display and editor
  */
-export function getValType(def, value) {
+
+import { UiSettingsType } from 'src/core/server/types';
+import { FieldSetting } from '../types';
+
+export function getValType(def: Partial<FieldSetting>, value?: any): UiSettingsType {
```

Comment:
Changing the return type from string to UiSettingsType is causing the current test failures.

Reply:
Allowing this to return `undefined` seems to allow everything to work. Will take a look at master to verify this is preexisting behavior and not something I introduced.

Reply:
Allowing this to return undefined seems to allow everything to work. Will take a look at master to verify this is preexisting behavior and not something I introduced.

Reply:
Verified that this is behavior on master in this pr - https://github.com/elastic/kibana/pull/55551